### ISSUE TITLE
GCC's libatomic.so not found. on Gentoo

### ISSUE THREAD

**Issue by idc77** [swgemu/Core3#9]
```
make -j8
mkdir -p build/unix
cd build/unix && cmake -DBUILD_IDL=ON -DRUN_GIT=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_CXX_FLAGS=""  ../.. && make
-- Using engine3 git submodule
-- CMAKE_MODULE_PATH: /home/swg/Core3/MMOCoreORB/cmake/Modules//home/swg/Core3/MMOCoreORB/utils/engine3/MMOEngine/build/cmake/Modules/
-- GCC's libatomic.so not found.
-- Try: 'sudo yum install libatomic' or 'sudo apt-get install libatomic1')
-- Building private engine
CMake Warning (dev) at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (jemalloc)
  does not match the name of the calling package (JEMALLOC).  This can lead
  to problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/Modules/FindJEMALLOC.cmake:44 (find_package_handle_standard_args)
  CMakeLists.txt:153 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- GCC's libatomic.so not found.
-- Try: 'sudo yum install libatomic' or 'sudo apt-get install libatomic1')
CMake Deprecation Warning at utils/googletest-release-1.10.0/CMakeLists.txt:4 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at utils/googletest-release-1.10.0/googlemock/CMakeLists.txt:45 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at utils/googletest-release-1.10.0/googletest/CMakeLists.txt:56 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Build type: RelWithDebInfo
-- Compiler cxx release with debug flags: -O2 -g -DNDEBUG
-- Compiler extra cxx flags: -Wall -Wextra -Wno-unused -Wno-sign-compare -Wno-multichar -Wno-unused-parameter -Wno-overloaded-virtual -Wno-missing-declarations -Wno-implicit-fallthrough -Wno-class-memaccess -Werror -march=native -fno-omit-
frame-pointer
-- Available targets: tags compdb update-engine3 core3 odb3 idl rebuild-idl
-- Configuring done (0.3s)
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
GCCLIBATOMIC_LIBRARY (ADVANCED)
    linked by target "testsuite3" in directory /home/swg/Core3/MMOCoreORB/utils/engine3/MMOEngine/src/testsuite3
    linked by target "core3" in directory /home/swg/Core3/MMOCoreORB/src
    linked by target "core3client" in directory /home/swg/Core3/MMOCoreORB/src/client

-- Generating done (0.2s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
make: *** [Makefile:121: build-cmake-with-idl] Error 1
```

I'm using gcc 12 and have tried 11.3.0, also 7.3.0

**Comment by LiveLaughLizard** [comment]
Supported operating systems, as far as I'm currently aware, are Ubuntu 20.04 and Debian 10. I think you'll be hard pressed to find any support outside of those at the moment.

**Comment by Hakry** [comment]
1sudo is correct. Current development setup guide is for WSL using Ubuntu 20.04 but you can also use Debian10.

**Comment by idc77** [comment]
I created a [thread on the Gentoo Forums ](https://forums.gentoo.org/viewtopic-p-8785831.html)and with the help of a friendly forum member named ts-plus there I was able to solve the Cmake find util not finding libatomic.so.

However a new problem arose. Core3 uses deprecated glibc functions which were removed in 2.34. I'm guessing Ubuntu 20.04 uses a glibc < 2.34, indeed [2.31-0ubuntu9](https://www.ubuntuupdates.org/package/core/focal/universe/base/glibc).

Since 20.04 is supported until 04/2030 I don't see this package moving away from the glibc <2.34 dependency anytime soon. Thus building and running on Gentoo is not possible, unless one limits their OS to use glibc <2.34.
